puppet/chassis.rb

```diff
@@ -102,16 +102,68 @@ def self.config
 	def self.install_extensions(config)
 		# Install extensions listed in config
 		if config["extensions"]
-			config["extensions"].each do |ext|
-				begin
-					repo = /^(git@|https:)/.match(ext) ? ext : 'https://github.com/' + ext
-					folder = @@dir + '/extensions/' + ext.split('/').last.gsub(/.git$/, '')
+			self.install_extensions_from_array(config["extensions"])
+		end
+
+		# For each of the extensions in our folder, read the extension config and
+		# install dependencies, etc for that extension.
+		Dir.foreach('extensions/') do |ext_folder|
+
+			# Bail out if we aren't a valid folder.
+			next if ext_folder == '.' or ext_folder == '..' or File.file?('extensions/' + ext_folder)
+
+			# If we don't have our config file, don't do anything.
+			next if not File.exists? 'extensions/' + ext_folder + '/chassis.yaml'
```

Comment:
🔢 Rather than `chassis.yaml` let's go with `config.yaml` instead. I've tested using that filename in a branch and it works fine! 🎉